### PR TITLE
Updating GetLastNetworkInference query and update old query

### PR DESCRIPTION
**Pull request by guilherme-brandao**
- `GetLastNetworkInference` was using `EpochLastEnded` as a reference to query the network inference, which is not optimal since this property is updated before the worker sends the inferences.

- Also removed a potentially unnecessary calculation in the `GetNetworkInferenceAtBlock` function. We're already calculating and setting the previous network combined loss and saving it during the losses insert from the reputer in the previous time step, so we just need to query it; we were calculating it again.

Ticket: https://linear.app/upshot/issue/ORA-1666/debug-getlatestnetworkinference-returning-empty-values-on-edgenet

### CODE REVIEW

File: x/emissions/keeper/inference_synthesis/network_losses.go [hunk 1]
```diff
@@ -210,43 +210,6 @@ func CalcNetworkLosses(
 	return output, nil
 }
 
-// Same as CalcNetworkLosses() but just returns the combined loss
-func CalcCombinedNetworkLoss(
```
Comment (relyt29) [inline x/emissions/keeper/inference_synthesis/network_losses.go:214]: is there a reason we're deleting this test, don't love deleting tests without a reason

File: x/emissions/keeper/inference_synthesis/network_losses_test.go [hunk 2]
```diff
@@ -61,144 +61,6 @@ func (s *InferenceSynthesisTestSuite) TestRunningWeightedAvgUpdate() {
 	}
 }
 
-func (s *InferenceSynthesisTestSuite) TestCalcCombinedNetworkLossSimpleCaseWithOneReputer() {
```
Comment (relyt29) [inline x/emissions/keeper/inference_synthesis/network_losses_test.go:64]: is there a reason we're deleting this test, don't love deleting tests without a reason

**Comment by relyt29** [comment]
I'm not super familiar with this part of the codebase, but to my untrained eye it looks fine

File: x/emissions/keeper/inference_synthesis/network_losses_test.go (hunk 2 above)
Comment (br4e) [reply x/emissions/keeper/inference_synthesis/network_losses_test.go:64]: Same, does this test break with the changes?

File: x/emissions/keeper/inference_synthesis/network_losses.go (hunk 1 above)
Comment (br4e) [reply x/emissions/keeper/inference_synthesis/network_losses.go:214]: this isn't a test though right?

File: x/emissions/keeper/inference_synthesis/network_inferences.go [hunk 3]
```diff
@@ -71,55 +69,131 @@ func GetNetworkInferencesAtBlock(
...
+				Epsilon:             moduleParams.Epsilon,
+				FTolerance:          moduleParams.FTolerance,
+				PNorm:               topic.PNorm,
+				CNorm:               moduleParams.CNorm,
+			},
 		)
 		if err != nil {
-			Logger(ctx).Warn(fmt.Sprintf("Error calculating network combined loss: %s", err.Error()))
-			return networkInferences, nil, infererWeights, forecasterWeights, nil
+			Logger(ctx).Warn(fmt.Sprintf("Error constructing network inferences builder topic: %s", err.Error()))
```
Comment (br4e) [inline x/emissions/keeper/inference_synthesis/network_inferences.go:100]: logger.warn here, but not in other places

File: x/emissions/keeper/inference_synthesis/network_inferences.go [hunk 4]
```diff
@@ -71,55 +69,131 @@ func GetNetworkInferencesAtBlock(
...
+	}
+	if len(inferences.Inferences) > 1 {
+		moduleParams, err := k.GetParams(ctx)
+		if err != nil {
+			return nil, nil, infererWeights, forecasterWeights, err
 		}
 		topic, err := k.GetTopic(ctx, topicId)
 		if err != nil {
-			Logger(ctx).Warn(fmt.Sprintf("Error getting topic: %s", err.Error()))
+			fmt.Printf("Error getting topic: %s", err.Error())
```
Comment (br4e) [inline x/emissions/keeper/inference_synthesis/network_inferences.go:170]: it looks like we are doing loger.warn in some places and fmt.printf in others? Let's just pick on and stick to it (I don't have a strong opinion, but I think logger.warn is what we want right?)

File: x/emissions/keeper/inference_synthesis/network_inferences.go [hunk 5]
```diff
@@ -71,55 +69,131 @@ func GetNetworkInferencesAtBlock(
...
+				FTolerance:          moduleParams.FTolerance,
+				PNorm:               topic.PNorm,
+				CNorm:               moduleParams.CNorm,
+			},
 		)
 		if err != nil {
-			Logger(ctx).Warn(fmt.Sprintf("Error calculating network combined loss: %s", err.Error()))
-			return networkInferences, nil, infererWeights, forecasterWeights, nil
+			Logger(ctx).Warn(fmt.Sprintf("Error constructing network inferences builder topic: %s", err.Error()))
+			return nil, nil, infererWeights, forecasterWeights, err
```
Comment (br4e) [inline x/emissions/keeper/inference_synthesis/network_inferences.go:101]: why can't we return network inferences here? we did above

File: x/emissions/keeper/inference_synthesis/network_inferences.go [hunk 6]
```diff
@@ -71,55 +69,131 @@ func GetNetworkInferencesAtBlock(
...
-			stakeAmount, err := k.GetStakeOnReputerInTopic(ctx, topicId, bundle.ValueBundle.Reputer)
-			if err != nil {
-				Logger(ctx).Warn(fmt.Sprintf("Error getting stake on reputer: %s", err.Error()))
-				return networkInferences, nil, infererWeights, forecasterWeights, nil
-			}
-			stakesByReputer[bundle.ValueBundle.Reputer] = stakeAmount
+		topic, err := k.GetTopic(ctx, topicId)
+		if err != nil {
+			Logger(ctx).Warn(fmt.Sprintf("Error getting topic: %s", err.Error()))
+			return networkInferences, nil, infererWeights, forecasterWeights, nil
```
Comment (br4e) [inline x/emissions/keeper/inference_synthesis/network_inferences.go:81]: we returned networkInferences here, so can we also return them below?

**Review by br4e — COMMENTED** [review]
For testing, can you just test using the existing test `TestGetLatestNetworkInferences`?

File: x/emissions/keeper/inference_synthesis/network_losses_test.go (hunk 2 above)
Comment (guilherme-brandao) [reply x/emissions/keeper/inference_synthesis/network_losses_test.go:64]: this is a test for the `CalcCombinedNetworkLoss` which is a function that I deleted

File: x/emissions/keeper/inference_synthesis/network_losses.go (hunk 1 above)
Comment (relyt29) [reply x/emissions/keeper/inference_synthesis/network_losses.go:214]: oh for some reason i thought it was in network_losses_test.go

**Pull request closed without merging**